warehouse/query-core/src/test/java/datawave/query/tables/async/event/VisitorFunctionTest.java

```diff
@@ -0,0 +1,292 @@
+        
+        verifyAll();
+        
+        Assert.assertNotEquals(chunk, updatedChunk);
+        String updatedQuery = updatedChunk.getOptions().getIterators().iterator().next().getOptions().get(QueryOptions.QUERY);
+        Assert.assertNotEquals(query, updatedQuery);
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("_List_"));
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("field = 'FIELD1'"));
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("values\":[\"a\",\"b\",\"c\",\"d\",\"e\"]"));
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("FIELD1 == 'x' || FIELD1 == 'y'"));
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("&& (field = 'FIELD2') && (params = '{\"values\":[\"a\",\"b\",\"c\",\"d\"]}')"));
+    }
+    
+    @Test
+    public void rangeOverTermThresholdTest() throws IOException, TableNotFoundException, URISyntaxException {
+        setupExpects();
+        
+        config.setCleanupShardsAndDaysQueryHints(false);
+        config.setBypassExecutabilityCheck(true);
+        config.setSerializeQueryIterator(false);
+        
+        Query mockQuery = createMock(Query.class);
+        config.setQuery(mockQuery);
+        EasyMock.expect(mockQuery.getId()).andReturn(new UUID(0, 0)).anyTimes();
+        EasyMock.expect(mockQuery.duplicate("testQuery1")).andReturn(mockQuery).anyTimes();
+        
+        // set thresholds
+        config.setMaxTermThreshold(1);
+        config.setMaxDepthThreshold(10);
+        config.setMaxOrExpansionFstThreshold(100);
+        config.setMaxOrExpansionThreshold(20);
+        config.setMaxOrRangeThreshold(2);
+        config.setMaxRangesPerRangeIvarator(50);
+        config.setMaxOrRangeThreshold(2);
+        
+        SessionOptions options = new SessionOptions();
+        IteratorSetting iteratorSetting = new IteratorSetting(10, "itr", QueryIterator.class);
+        String query = "((_Bounded_ = true) && (FIELD1 > 'a' && FIELD1 < 'y')) || ((_Bounded_ = true) && (FIELD1 > 'c' && FIELD1 < 'z'))";
+        iteratorSetting.addOption(QueryOptions.QUERY, query);
+        options.addScanIterator(iteratorSetting);
+        
+        ScannerChunk chunk = new ScannerChunk(options, Collections.singleton(new Range()));
+        
+        replayAll();
+        
+        function = new VisitorFunction(config, helper);
+        ScannerChunk updatedChunk = function.apply(chunk);
+        
+        verifyAll();
+        
+        Assert.assertNotEquals(chunk, updatedChunk);
+        String updatedQuery = updatedChunk.getOptions().getIterators().iterator().next().getOptions().get(QueryOptions.QUERY);
+        Assert.assertNotEquals(query, updatedQuery);
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("_List_"));
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("field = 'FIELD1'"));
+        Assert.assertTrue(updatedQuery, updatedQuery.contains("ranges\":[[\"(a\",\"z)\"]"));
+    }
+}
```

Comment:
Is the addition of the VisitorFunctionTest sufficient or should a test be added for the PushdownLargeFieldedListsVisitor?